pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -229,6 +234,29 @@ func (r *ReconcileToolchainStatus) notificationCheck(reqLogger logr.Logger, tool
 	return nil
 }
 
+func (r *ReconcileToolchainStatus) restoredCheck(reqLogger logr.Logger, toolchainStatus *toolchainv1alpha1.ToolchainStatus) error {
+	// If the current ToolchainStatus:
+	// a) Was not ready before,
+	// b) notification was sent out due prolonged not ready status, and
+	// c) status is now restored
+	// send a notification to the admin mailing list
+	c, found := condition.FindConditionByType(toolchainStatus.Status.Conditions, toolchainv1alpha1.ConditionReady)
+	if found && c.Status == corev1.ConditionFalse {
+		c, found = condition.FindConditionByType(toolchainStatus.Status.Conditions, toolchainv1alpha1.ToolchainStatusUnreadyNotificationCreated)
+		if found && c.Status == corev1.ConditionTrue {
```

Comment:
Let's add `IsFalse(conditions []toolchainv1alpha1.Condition, conditionType toolchainv1alpha1.ConditionType) bool` next to https://github.com/codeready-toolchain/toolchain-common/blob/91c4f9716eff40a84239948f1c13e4f477146ce4/pkg/condition/condition.go#L63
so we can use it here as:
```suggestion
	if condition.isFalse(toolchainStatus.Status.Conditions, toolchainv1alpha1.ConditionReady) &&
	   condition.isTrue(toolchainStatus.Status.Conditions, toolchainv1alpha1.ToolchainStatusUnreadyNotificationCreated) {
```